### ISSUE TITLE
Black Beast Chest bug?

### ISSUE THREAD

**Issue by RevanXVIV** [jangler/oracles-randomizer#159]
Ok so my members card is behind black beast chest and I'm doing an entrance randomizer, the second I opened the black beast chest and the card appeared, the game kept moving but the text never appeared so I'm stuck staring at the card forever. Is this a known bug? Is there a fix?

**Comment by Ahrio76** [comment]
i'm having the same issue but with the boomerang, both of them, so now I can't complete my seed..

**Comment by jangler** [comment]
I think I know what went wrong here, but not why it went wrong. There aren't any special properties which that chest/room has, as far as I'm aware.

Can someone who's had this problem post the top few lines of their spoiler log, so I can see the exact version, settings, and seed? I'll try to reproduce it and see what's happening in the debugger.

**Comment by Ahrio76** [comment]
oracles randomizer 1.6.1_entrance-webpatch-d232c78
generated 2023-10-22T19:27:24+01:00
seed: bba20000
sha-1 sum: e4710fe6fc2b1cd5a4cf3ece868f3e27de9538ec
difficulty: normal

**Comment by jangler** [comment]
After asking about this in Discord, it looks like this is probably caused by sprite space expansion for the Nayru player sprite. Custom player sprites aren't a feature of this randomizer per se, but there is a bit of code in this randomizer that makes the Nayru player sprite *possible* for an external patcher (like the web patcher) to do. Apparently there was a bug with it that Stewmat contributed a fix for years ago, but the entrance randomizer fork hasn't been updated in a long time and so it hasn't merged that fix from this codebase.

My understanding is that the corresponding crash/softlock only happens when using the Nayru player sprite. Were you using that sprite for your playthrough?

**Comment by Ahrio76** [comment]
yes, the red one

**Comment by jangler** [comment]
@ReggX or @vinheim3 - possible to get a new release of the entrance randomizer? Merging [4.0.A](https://github.com/jangler/oracles-randomizer/releases/tag/4.0.A) into entrance would fix the problem without having to deal with merging all the multiworld changes, in case that's a problem.

**Comment by ReggX** [comment]
I'll be honest. I haven't looked at the code in years. I assume you are talking about this one relevant commit https://github.com/jangler/oracles-randomizer/commit/445fa3555f89e8560969e491cb4642c6305d0e8d changing eob.yaml.

If so, then that should be managable.

**Comment by ReggX** [comment]
Scratch that. I have neither my build toolchain ready nor committer privileges on @vinheim3, so I can't draft a new release.

I still pushed the fix to https://github.com/ReggX/oracles-randomizer/commit/687e02a66369827ef5019dae68387f21a4d5197c but the entrance rando branch should be considered abandoned in its current form. Anybody is free to fork and continue working on it if they want to though.

**Comment by jangler** [comment]
Yes, that's the relevant commit. Maybe I'll see if I can put together a new entrance rando release for cemulate sometime, then.
